src/test/java/edu/harvard/iq/dataverse/util/cache/RateLimitUtilTest.java

```diff
@@ -0,0 +1,128 @@
+package edu.harvard.iq.dataverse.util.cache;
+
+import edu.harvard.iq.dataverse.authorization.users.AuthenticatedUser;
+import edu.harvard.iq.dataverse.authorization.users.GuestUser;
+import edu.harvard.iq.dataverse.authorization.users.User;
+import edu.harvard.iq.dataverse.util.SystemConfig;
+import edu.harvard.iq.dataverse.util.cache.RateLimitUtil;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.junit.jupiter.MockitoSettings;
+import org.mockito.quality.Strictness;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.LENIENT)
+public class RateLimitUtilTest {
+
+    private SystemConfig mockedSystemConfig;
+
+    static final String settingJson = "{\n" +
```

Comment:
**suggestion**: The readability and maintainability of this JSON test example would benefit greatly from Java 15+ text blocks. https://www.baeldung.com/java-text-blocks Maybe give it a try?

Reply:
Updated to use new code block